src/lib/index.js

```diff
@@ -616,3 +616,41 @@ lib.numSeparate = function(value, separators) {
 
     return x1 + x2;
 };
+
+/**
+ * Interleaves separate trace updates (frames) into a restyle command.
+ * Attributes not specified in both traces are set to `undefined` so that
+ * they are not altered by restyle. Object paths are *not* expanded.
+ *
+ * @example
+ * lib.interleaveTraceUpdates([{x: [1]}, {x: [2]}])
+ * // returns {x: [[1], [2]]}
+ *
+ * @param   {array} traces       the trace updates to be interleaved
+ *
+ * @return  {object}    an object contianing the interleaved properties
+ */
+lib.interleaveTraceUpdates = function(traces) {
+    var i, j, prop;
+    var n = traces.length;
+    var output = {};
+
+    for(i = 0; i < traces.length; i++) {
+        for(prop in traces[i]) {
```

Comment:
non- :no_entry_sign: , but we usually loop over object keys by first  `var keys = Object.keys(traces)` and then looping of `keys`.